internal/s3/external.go

```diff
@@ -0,0 +1,203 @@
+package s3
+
+import (
+	"context"
+	"fmt"
+
+	methodaws "github.com/Method-Security/methodaws/generated/go"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+)
+
+func getAWSRegions() []string {
+	resolver := endpoints.DefaultResolver()
+	partitions := resolver.(endpoints.EnumPartitions).Partitions()
+
+	var regions []string
+	for _, p := range partitions {
+		for region := range p.Regions() {
+			regions = append(regions, region)
+		}
+	}
+
+	return regions
+}
+
+func bucketExists(ctx context.Context, region string, bucketName string) (bool, error) {
+	// Create a custom AWS config with anonymous credentials
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion(region),
+		config.WithCredentialsProvider(aws.AnonymousCredentials{}),
+	)
+	if err != nil {
+		return false, fmt.Errorf("error loading AWS config: %v", err)
+	}
+
+	// Create an S3 client
+	client := s3.NewFromConfig(cfg)
+
+	// Call HeadBucket operation
+	_, err = client.HeadBucket(ctx, &s3.HeadBucketInput{
+		Bucket: aws.String(bucketName),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "NotFound":
+				return false, nil
+			case "Forbidden":
+				return true, nil
+			default:
+				return false, fmt.Errorf("error checking bucket: %v", err)
+			}
+		}
+		return false, fmt.Errorf("error checking bucket: %v", err)
+	}
+
+	// If there's no error, the bucket exists
+	return true, nil
+}
+
+// externalEnumerateS3Region enumerates a single public facing S3 bucket in a specific region.
+// If the bucket does not exist, it will return an unmodified report (with potential new errors).
+func externalEnumerateS3Region(ctx context.Context, report methodaws.ExternalS3Report, bucketName string, region string) methodaws.ExternalS3Report {
```

Comment:
Can we please break this up into a few smaller helper functions? This is doing quite a bit.